### PR TITLE
Trap focus in modal (#310)

### PR DESCRIPTION
**Pull request by monteith**
- [ ] Bug fix 
- [x] New feature 
- [ ] Chore 
- [ ] Breaking change 
- [x] There is an open issue which this change addresses
- [x] I have read the **[CONTRIBUTING](./CONTRIBUTING.md)** document.
- [x] My commits follow the [Git Commit Guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits)
- [x] My code follows the code style of this project.
- [] My change requires a change to the documentation.
- - [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

I was unable to implement a test with enzyme. Focus on elements doesn't seem to advance with tabs in a mounted component. I could use some help understanding how to accomplish this test.

#310

### CODE REVIEW

File: src/Modal.js [hunk 1]
```diff
@@ -182,6 +184,31 @@ class Modal extends React.Component {
     }
   }
 
+  handleTab(e) {
+    if (e.which !== 9) return;
+
+    const focusableChildren = this._element.querySelectorAll(focusableElements.join(', '));
+    const currentFocus = document.activeElement;
```
Comment (TheSharpieOne) [inline src/Modal.js]: `document.activeElement` can have issues in IE, specially within iframes.
I just ran into it the other day, https://github.com/Availity/react-block-ui/issues/18

File: src/utils.js [hunk 2]
```diff
@@ -260,3 +260,16 @@ export function addMultipleEventListeners(_els, handler, _events) {
...
+  'a[href]',
+  'area[href]',
+  'input:not([disabled])',
+  'select:not([disabled])',
+  'textarea:not([disabled])',
+  'button:not([disabled])',
+  'object',
+  'embed',
+  '*[tabindex]:not(.modal)',
+  '*[contenteditable]',
```
Comment (TheSharpieOne) [inline src/utils.js]: Probably also want `'audio[controls]'`, `'video[controls]'`, and to account for `contenteditable="false"` with `'[contenteditable]:not([contenteditable="false"])'`

Also, do you need the `*` in the selectors?

File: src/utils.js [hunk 3]
```diff
@@ -260,3 +260,16 @@ export function addMultipleEventListeners(_els, handler, _events) {
     });
   };
 }
+
+export const focusableElements = [
+  'a[href]',
+  'area[href]',
+  'input:not([disabled])',
```
Comment (TheSharpieOne) [inline src/utils.js]: Need to also account for `type="hidden"`

File: src/Modal.js (hunk 1 above)
Comment (monteith) [reply src/Modal.js]: Wrapped this in a try/catch, defaulting to the first focusable child. Tested on IE10, IE11 on win7 and MSEDGE on win10. Let me know if I missed a version.

File: src/utils.js (hunk 2 above)
Comment (monteith) [reply src/utils.js]: Maybe not. Personal preference. Removed 👍

File: src/utils.js (hunk 3 above)
Comment (monteith) [reply src/utils.js]: 👍

**Pull request merged**